app/Jobs/SendEmptyWikiNotificationsJob.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace App\Jobs;
+
+use App\Notifications\EmptyWikiNotification;
+use App\Wiki;
+use App\WikiNotificationSentRecord;
+use Carbon\CarbonImmutable;
+use Illuminate\Contracts\Queue\ShouldBeUnique;
+use Illuminate\Support\Facades\Log;
+
+class SendEmptyWikiNotificationsJob extends Job implements ShouldBeUnique
+{
+    public function handle (): void
+    {
+        $wikis = Wiki::with(['wikiLifecycleEvents'])->get();
+
+        foreach ($wikis as $wiki) {
+            try {
+                if ($this->checkIfWikiIsOldAndEmpty($wiki)) {
+                    $this->sendEmptyWikiNotification($wiki);
+                }
+            } catch (\Exception $exception) {
+                Log::error(
+                    'Failure processing wiki '.$wiki->getAttribute('domain').' for EmptyWikiNotification check: '.$exception->getMessage()
+                );
+                $this->fail();
+            }
+        }
+    }
+
+    public function checkIfWikiIsOldAndEmpty(Wiki $wiki)
+    {
+        // Calculate how many days has passed since the wiki instance was first created
+        $emptyDaysThreshold = config('wbstack.wiki_empty_notification_threshold');
+        $createdAt = $wiki->created_at;
+        $now = CarbonImmutable::now();
+        $emptyWikiDays = $createdAt->diffInDays($now);
+
+        $firstEdited = $wiki->wikiLifecycleEvents->first_edited;
+
+        $emptyWikiNotificationCount = WikiNotificationSentRecord::where([
+            'wiki_id' => $wiki->id,
+            'notification_type' => EmptyWikiNotification::TYPE
+        ])->count();
+
+        if (
+            $firstEdited == null &&
+            $emptyWikiDays >= $emptyDaysThreshold &&
+            $emptyWikiNotificationCount == 0
+        ) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public function sendEmptyWikiNotification (Wiki $wiki): void
+    {
+        $wikiManagers = $wiki->wikiManagers()->get();
+
+        foreach($wikiManagers as $wikiManager) {
+            $wiki->wikiNotificationSentRecords()->create([
```

Comment:
Like this, we'd also create a record when the notfiy call below fails immediately. Should we create the record _after_ calling `notify`?

Reply:
That's a great point but I'm not sure if that is the case since the `notify` method appears to dispatch the sending, but I'm not entirely sure how it all works in detail. My thinking was the other way around: prevent a case where we send a notification but don't keep a record, resulting in potentially spamming a user. But now I'm not sure how to prevent both cases